mudz.Core/Domain/GameEngine/HiveMind.cs

```diff
@@ -61,7 +61,7 @@ public Grid World
 
                 _world = new Grid()
                 {
-                    Rooms = new Dictionary<RoomKey, RoomContent>(),
+                    Rooms = new RoomCollection(),
```

Comment:
The intent is to provide some easy exstensibility without having to use extension methods across the Dictionary.